### ISSUE TITLE
NuGet.Config location varies by Mono/dotnet rather than by OS distribution

### ISSUE THREAD

**Issue by jcansdale** [NuGet/docs.microsoft.com-nuget#1878]
The following seems to vary depending on whether you're using `mono nuget` or `dotnet nuget`. 

```
Mac/Linux: ~/.config/NuGet/NuGet.Config or ~/.nuget/NuGet/NuGet.Config (varies by OS distribution)
```

`dotnet nuget` uses  `~/.nuget/NuGet/NuGet.Config`
`nuget` (via Mono) uses `~/.config/NuGet/NuGet.Config`

This is very confusing!

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 370d9d91-9f35-79aa-3ace-7b77436e988f
* Version Independent ID: 3da4591a-3d3d-c07a-6672-97964de13b14
* Content: [Common NuGet configurations](https://docs.microsoft.com/en-us/nuget/consume-packages/configuring-nuget-behavior#feedback)
* Content Source: [docs/consume-packages/configuring-nuget-behavior.md](https://github.com/NuGet/docs.microsoft.com-nuget/blob/master/docs/consume-packages/configuring-nuget-behavior.md)
* Product: **nuget**
* GitHub Login: @karann-msft
* Microsoft Alias: **karann**

**Comment by anangaur** [comment]
I think with .NET 5, we would also need to decide upon one location. 
/cc: @karann-msft @aortiz-msft @nkolev92

**Comment by anangaur** [comment]
@chgill-MSFT Can you check the behavior on Linux and Mac for Mono and update the docs?

**Comment by jcansdale** [comment]
@anangaur it looks like what is going on is this:

```cs
            // Used by .NET, Mono and .NET Core on Windows
            var config1 = Path.Combine(
                Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData),
                "NuGet", "NuGet.Config"
            );

            // Used by .NET Core on Mac and Linux
            var config2 = Path.Combine(
                Environment.GetFolderPath(Environment.SpecialFolder.UserProfile), ".nuget",
                "NuGet", "NuGet.Config"
            );
```

I initially assumed the `Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData)` based location would be used everywhere, but it turns out this can return `string.Empty` on _some_ Linux environments. When used inside VS Online it returns a path but when used inside GitHub Actions it returns `string.Empty`.

**Comment by nkolev92** [comment]
@jcansdale Can you please let us know how you determined that? 

My quick investigation is pointing to something different.

The entry point for determining the default config location is: 

https://github.com/NuGet/NuGet.Client/blob/024a759c126123d9a36569b3173d80011ee8466b/src/NuGet.Core/NuGet.Common/PathUtil/NuGetEnvironment.cs#L19

The argument is [NuGetFolderPath.UserSettingsDirectory](https://github.com/NuGet/NuGet.Client/blob/024a759c126123d9a36569b3173d80011ee8466b/src/NuGet.Core/NuGet.Common/PathUtil/NuGetFolderPath.cs#L10)

https://github.com/NuGet/NuGet.Client/blob/024a759c126123d9a36569b3173d80011ee8466b/src/NuGet.Core/NuGet.Common/PathUtil/NuGetEnvironment.cs#L46-L49

On .NET Core the check is: 

https://github.com/NuGet/NuGet.Client/blob/024a759c126123d9a36569b3173d80011ee8466b/src/NuGet.Core/NuGet.Common/PathUtil/NuGetEnvironment.cs#L129-L137

So APPDATA on Windows and the Home variable on linux/mac. 

On in VS/nuget.exe and nuget.exe on mono it is: 

https://github.com/NuGet/NuGet.Client/blob/024a759c126123d9a36569b3173d80011ee8466b/src/NuGet.Core/NuGet.Common/PathUtil/NuGetEnvironment.cs#L196-L198

Specifically `Environment.SpecialFolder.ApplicationData`. 

tl;dr. 

* Windows is consistent, it's always AppData. 
* nuget.exe on mono + dotnet.exe on linux/mac will be different because the first one is the Environment.SpecialFolder.ApplicationData location that the mono runtime returns and the later is the home variable.

**Comment by jcansdale** [comment]
@nkolev92,

Thanks for the links, I hadn't seen the actual source. 😄 

> Windows is consistent, it's always AppData.

You could also say that Windows and Mono is consistent, always AppData. I was surprised to find `dotnet nuget` using a different folder on on linux/mac. This is unfortunate because `dotnet nuget`currently lacks a `config` sub-command, so people will be inclined to use `mono nuget.exe config` instead. By default this will update the wrong `NuGet.Config` file. 😢 

(BTW, I work in Developer Support at GitHub and have seen a few tickets about this when users are configuring GitHub Packages)

> nuget.exe on mono + dotnet.exe on linux/mac will be different because the first one is the Environment.SpecialFolder.ApplicationData location that the mono runtime returns and the later is the home variable.

Yes, this matches what I've observed. 👍 

Cheers, Jamie.

**Comment by nkolev92** [comment]
> This is unfortunate because dotnet nugetcurrently lacks a config sub-command, so people will be inclined to use mono nuget.exe config instead. By default this will update the wrong NuGet.Config file

We are actively working on closing that gap. In fact it's something we are fixing in the next release of the SDK, https://github.com/NuGet/Home/issues/4126.

**Comment by nkolev92** [comment]
Duplicate of https://github.com/NuGet/Home/issues/4413

**Comment by pMaske** [comment]
I am also struggling how to change the default nuget.exe to be used when I run my build script on mac. 
I install nuget.exe to a custom folder on Mac and want to use the one that I installed instead of using nuget.exe from mono.
I also added custom location to the PATH environment variable but it still uses the one from Mono. Is there a way to change which nuget.exe to use on Mac?
I also created an issue # 2009 yesterday with my code but thought will ask here also since in case some has any insight on this.
